### PR TITLE
DRYD-943: Add Transport Procedure

### PR DESCRIPTION
**Pull request by mikejritter**
Resolves: https://collectionspace.atlassian.net/browse/DRYD-943

* Adds ui components for transport procedure

Note: I wasn't really sure what to do for the title, so I only used the `transportReferenceNumber`.

### CODE REVIEW

File: src/plugins/recordTypes/transport/optionLists.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        id: 'option.transportMethodType.Exclusive-use truck',
+        defaultMessage: 'exclusive-use truck',
+      },
+      'Expedited use freight': {
+        id: 'option.transportMethodType.Expedited use freight',
+        defaultMessage: 'expedited use freight',
+      },
+      'LOFO freight': {
+        id: 'option.transportMethodType.LOFO freight',
+        defaultMessage: 'lofo freight',
```
Comment (ray-lee) [inline src/plugins/recordTypes/transport/optionLists.js:41]: It looks like "LOFO" is normally all caps... let's go with that.

File: src/plugins/recordTypes/transport/optionLists.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      'Common carrier',
+      'Exclusive-use truck',
+      'Expedited use freight',
+      'LOFO freight',
+      'Mail',
+      'Non-commercial carrier',
+      'Ocean freight',
+      'Passenger aircraft',
+      'Shuttle service',
+    ],
```
Comment (ray-lee) [inline src/plugins/recordTypes/transport/optionLists.js:17]: For static option lists, the best practice is to make the data values identical to the default message string, including case. That way you get exactly the same value if you're querying the database or REST API as you do in the UI, as long as you haven't overridden the default message. So it's best to lowercase all the above values (except for "LOFO freight" -- see comment below).

**Review by ray-lee — CHANGES_REQUESTED** [review]
Mike, this looks good. Just a couple comments. Thanks!

File: src/plugins/recordTypes/transport/fields.js [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+              },
+            }),
+            view: {
+              type: TextInput,
+            },
+          },
+        },
+        numberOfCrates: {
+          [config]: {
+            dataType: DATA_TYPE_FLOAT,
```
[the remark this answers is not in the document]
Comment (mikejritter) [reply src/plugins/recordTypes/transport/fields.js:241]: It looks like I missed the `DATA_TYPE_INT` in general - certainly would make more sense for this field. Looking back at the changes I made to the application I used `datatype=float` as well in `base-procedure-transport.xml`. Do you want me to push an update to update it there as well?

[the remark this answers is not in the document]
Comment (ray-lee) [reply src/plugins/recordTypes/transport/fields.js]: Oh, I missed that too. Don't worry about it, I'll change it while I'm in there.

**Pull request merged**